src/algebra/algebra/basic.lean

```diff
@@ -1433,15 +1441,29 @@ instance restrict_scalars.module_orig [semiring A] [add_comm_monoid M] [I : modu
 variables [comm_semiring R] [semiring A] [algebra R A]
 variables [add_comm_monoid M] [module A M]
 
-/--
-When `M` is a module over a ring `A`, and `A` is an algebra over `R`, then `M` inherits a
-module structure over `R`.
-
-The preferred way of setting this up is `[module R M] [module A M] [is_scalar_tower R A M]`.
--/
+/-- When `M` is a module over a semiring `A`, and `A` is an algebra over `R`, then `M` inherits a
+module structure over `R`. -/
 instance : module R (restrict_scalars R A M) :=
 module.comp_hom M (algebra_map R A)
 
+namespace restrict_scalars
+
+/-- `restrict_scalars` is an equivalence of modules over the semiring `A`. -/
+@[nolint unused_arguments]
```

Comment:
I think this lint warning is valid and should be fixed. You don't need `A` to be an `R`-algebra for this to make sense; `R` is completely irrelevant to this definition.

Reply:
OK, done.